mne/io/curry/curry.py

```diff
@@ -0,0 +1,317 @@
+                                ["meg", "eeg", "misc"]])
+
+    sensors = _read_curry_lines(fname_base + LABEL_FILE_EXTENSION[curry_vers],
+                                ["SENSORS" + CHANTYPES[key] for key in
+                                 ["meg", "eeg", "misc"]])
+
+    all_chans = list()
+    for key in ["meg", "eeg", "misc"]:
+        for ind, chan in enumerate(labels["LABELS" + CHANTYPES[key]]):
+            ch = {"ch_name": chan,
+                  "unit": unit_dict[key],
+                  "kind": FIFFV_CHANTYPES[key]}
+            if key in ("meg", "eeg"):
+                loc = sensors["SENSORS" + CHANTYPES[key]][ind]
+                ch["loc"] = np.array(loc, dtype=float)
+
+            all_chans.append(ch)
+
+    ch_names = [chan["ch_name"] for chan in all_chans]
+    info = create_info(ch_names, sfreq)
+
+    for ind, ch_dict in enumerate(info["chs"]):
+        ch_dict["kind"] = all_chans[ind]["kind"]
+        ch_dict['unit'] = SI_UNITS[all_chans[ind]['unit'][1]]
+        ch_dict['cal'] = SI_UNIT_SCALE[all_chans[ind]['unit'][0]]
+        if ch_dict["kind"] in (FIFF.FIFFV_MEG_CH,
+                               FIFF.FIFFV_EEG_CH):
+            ch_dict["loc"] = all_chans[ind]["loc"]
+
+    return info, n_samples, is_ascii
+
+
+def _read_events_curry(fname):
+    """Read events from Curry event files.
+
+    Parameters
+    ----------
+    fname : str
+        Path to a curry event file with extensions .cef, .ceo,
+        .cdt.cef, or .cdt.ceo
+
+    Returns
+    -------
+    events : ndarray, shape (n_events, 3)
+        The array of events.
+    """
+    check_fname(fname, 'curry event', ('.cef', '.cdt.cef'),
+                endings_err=('.cef', '.cdt.cef'))
+
+    events_dict = _read_curry_lines(fname, ["NUMBER_LIST"])
+    # The first 3 column seem to contain the event information
+    curry_events = np.array(events_dict["NUMBER_LIST"], dtype=int)[:, 0:3]
+
+    return curry_events
+
+
+def _read_annotations_curry(fname, sfreq='auto'):
+    r"""Read events from Curry event files.
+
+    Parameters
+    ----------
+    fname : str
+        The filename.
+    sfreq : float | 'auto'
+        The sampling frequency in the file. If set to 'auto' then the
+        ``sfreq`` is taken from the respective info file of the same name with
+        according file extension (\*.dap for Curry 7; \*.cdt.dpa for Curry8).
+         So data.cef looks in data.dap and data.cdt.cef looks in data.cdt.dpa.
+
+    Returns
+    -------
+    annot : instance of Annotations | None
+        The annotations.
+    """
+    events = _read_events_curry(fname)
+
+    if sfreq == 'auto':
+        fname_base, ext = fname.split(".", maxsplit=1)
+        curry_vers = _get_curry_version(ext)
+        with open(fname_base + INFO_FILE_EXTENSION[curry_vers]) as fid:
+            for line in fid:
+                if ('SampleFreqHz' or 'SAMPLE_FREQ_HZ') in line:
+                    sfreq = float(line.split("=")[1])
```

Comment:
@massich @agramfort 

if we replace this with:
```python
    if sfreq == 'auto':
        fname_base, ext = fname.split(".", maxsplit=1)
        curry_vers = _get_curry_version(ext)
        with open(fname_base + INFO_FILE_EXTENSION[curry_vers]) as fid:
            for line in fid:
                if ('SampleFreqHz' or 'SAMPLE_FREQ_HZ') in line:
                    sfreq = float(line.split("=")[1])
                elif ('SampleTimeUsec' or 'SAMPLE_TIME_USEC') in line:
                    time_step = float(line.split("=")[1]) * 1e-6

        if (sfreq == 0) and (time_step != 0):
            sfreq = 1. / time_step
```
It will also be able to automatically set a sfreq of 0 to 1/time_step.
However, to test it, we would have to introduce yet another file into the mne-testing-data :/

Reply:
so is it possible that in the end sfreq is 0 on read? this should not happen I would say. This should crash

is it possible that both SampleFreqHz and SampleTimeUsec are present with contradictory values?

Reply:
If an sfreq of 0 is passed to `read_annotations` or `_read_annotations_curry`, it will give a ZeroDivisionError in the next line `onset = events[:, 0] / sfreq`.
So this won't pass with wrong results.

> is it possible that both SampleFreqHz and SampleTimeUsec are present with contradictory values?

I honestly can't tell you. EEGLAB's loadcurry implements something similar to ` if (sfreq == 0) and (time_step != 0):
            sfreq = 1. / time_step`, so i added it here too. However, from the 3 files that Michael Wagner converted for testing, sfreq was always defined.